app/__tests__/popup.test.tsx

```diff
@@ -0,0 +1,103 @@
+///<reference path='./../../node_modules/@types/enzyme/index.d.ts'/>
+///<reference path='./../../node_modules/@types/jasmine/index.d.ts'/>
+///<reference path='./../../node_modules/@types/enzyme-adapter-react-15/index.d.ts'/>
+///<reference path='./../../node_modules/@types/jest/index.d.ts'/>
+import * as React from 'react';
+import { shallow } from 'enzyme';
+import {Popup} from './../popup';
+
+import * as MessageService from '../message_service';
```

Comment:
I’d suggest writing tests in JS. Useless boilerplate of TS gets in the way of understanding the test itself, IMO